TOY-47 설정 화면 및 Firebase Auth 로그인 기능 구현

pby2017 opened the pull request:
코드 리뷰 부탁드립니다.

1. Preference support library를 사용해서 설정 화면 구현
2. Firebase auth library를 사용해서 이메일/비밀번호 가입 및 로그인 기능 구현

merge 전 
MainActivity에서 설정화면으로 이동하는 버튼을 제거하고
메인 다이어리에서 설정화면으로 이동하도록 수정하겠습니다.

pby2017 committed:
TOY-47 설정 화면 및 구글 인증 가입/로그인 구현

pby2017 committed:
TOY-47 파일명 및 변수명 수정 (login -> sign)

pby2017 committed:
TOY-47 LoginCallback 인터페이스 분리

pby2017 committed:
TOY-47 입력 공백 처리 수정

pby2017 committed:
TOY-47 아이콘 추가

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/settings/LoginPreferenceDialogFragmentCompat.java:
```diff
@@ -0,0 +1,88 @@
+    private EditText emailEdit;
+    private EditText passwordEdit;
+    private TextView showSignFailText;
+    private ProgressBar signProgress;
```
BaseActivity에 구현해놓고 가져다 쓰는 형태로 하는게 좋겠어요.

pby2017 replied:
네
반영했습니다.

bonchangkoo reviewed (approved):
코드 확인했습니다.

hyejin830 commented:
로그인 후 로그인 상태인 것을 인지 할 수 있도록 반영해야 할 것 같고, 
`로그인` 텍스트가 `로그아웃`으로 변경되어야 할 것 같습니다. 
그리고 로그아웃 기능도 있어야 할 것같구요!

또 로그인 할 때, 회원가입도 아이디/비밀번호 입력하면 된다는 것을 알 수 있게 작성 해 놓으시면 더 좋을 것 같습니다~ 처음에 가입을 어떻게 하는지 잘 모르겠습니다

hyejin830 commented:
확인 완료했습니다

pby2017 committed:
TOY-47 피드백 반영

pby2017 committed:
TOY-47 Merge branch 'develop' into TOY-47-setting-auth

# Conflicts:
#	app/build.gradle
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/BaseActivity.kt
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/diary/main/DiariesActivity.java
#	app/src/main/res/values/strings.xml

pby2017 commented:
> 로그인 후 로그인 상태인 것을 인지 할 수 있도록 반영해야 할 것 같고,
> `로그인` 텍스트가 `로그아웃`으로 변경되어야 할 것 같습니다.
> 그리고 로그아웃 기능도 있어야 할 것같구요!
> 
> 또 로그인 할 때, 회원가입도 아이디/비밀번호 입력하면 된다는 것을 알 수 있게 작성 해 놓으시면 더 좋을 것 같습니다~ 처음에 가입을 어떻게 하는지 잘 모르겠습니다

네

반영했습니다.
